lib/gemstash/gem_source/private_source.rb

```diff
@@ -67,7 +67,15 @@ def serve_info(name)
       end
 
       def serve_marshal(id)
-        halt 403, "Not yet supported"
+        gem_full_name = id.sub(/\.gemspec\.rz\z/, "")
+        gem = storage.resource(gem_full_name)
+        halt 404 unless gem.exist?
+        gem.load
+        halt 403, "That gem has been yanked" unless gem.properties[:indexed]
+        spec = spec_storage.resource(gem_full_name)
+        halt 404 unless spec.exist?
+        content_type "application/octet-stream"
+        spec.load.content
       end
```

Comment:
Yep, let's avoid the early abstraction, agreed.

Yes, you don't seem to be using the properties for anything else that is not the "indexed" and given that it is just one thing it will add little to no value, so, again, let's move on and then take a look back to see if it makes any sense to extract anything later.